### DISCUSSION TITLE
copilot chat

### DISCUSSION THREAD

**Discussion by locomotive1995** [community/community#129135] [Copilot Conversations] [answered]
### Select Topic Area

Question

### GitHub Feature Area

Actions

### Body

i  think i will use copilot for free just 1 month
i know it is useful to get a code but is it possible to chat in copilot?
i mean any questions like coding, system, sas, r ...

**Comment by FreddyMSchubert** [comment, marked as answer]
The code completions are convenient. But if you actually want to properly setup some clean logic, Copilot won't be of much help in the long run. It's great for small, self contained functions (that have good naming so i can create them).
The chat is usually rather unhelpful. I much prefer openai gpt 4, which is still the best, or the free 4o, if you have access to that. When I'm very tired the chat will be helpful but usually I'm better of thinking for myself. But the chat does - exist.
I'd recommend anyone to try it for a month, but buying it without getting it indirectly through e.g. the github education thing I wouldn't recommend.
